src/AWS/Lambda/Events/S3/DeleteEvent.hs

```diff
@@ -0,0 +1,28 @@
+{-|
+Module      : AWS.Lambda.Events.S3.DeleteEvent
+Description : Data types for working with S3 delete events.
+Copyright   : (c) Nike, Inc., 2019
+License     : BSD3
+Maintainer  : nathan.fairhurst@nike.com, fernando.freire@nike.com
+Stability   : stable
+-}
+
+module AWS.Lambda.Events.S3.DeleteEvent (
+  module AWS.Lambda.Events.S3.BaseEvent,
+  S3DeleteObject(..),
+  DeleteRecords
+) where
+
+import Data.Aeson   (FromJSON (..))
+import Data.Text    (Text)
+import GHC.Generics (Generic)
+import AWS.Lambda.Events.S3.BaseEvent
+
+data S3DeleteObject = S3DeleteObject {
```

Comment:
I didn't want to force `Maybe` onto `etag` and `size` fields in the `PutObject` so this is the approach I came up with. An unfortunate side effect of this approach is that the `DeleteRecords` alias will _also_ parse `PutRecords` because Aeson ignores the extra fields. Not sure if we want to force that constraint in this implementation, since I imagine the most common use case will be to listen for a single event type (could be wrong though).

Reply:
Are we sure the two can't mix?  It might be interesting to do the alternative pattern here.  So it would decode into a `Put x | Delete y`.  Seems strange that there's no way to distinguish them otherwise, and two distinct `Maybe`s that are Nothing or Just at the same time doesn't seem like an ideal encoding--much agreed there.

Reply:
Not sure at all. I'll spin up a test with put/delete in the same function and see what shows up! Alternative makes it pretty straightforward too, I like that idea!

Reply:
I am much happier with the current implementation :) Thanks!